README.adoc

```diff
@@ -1,5 +1,174 @@
+We use the generator in these meta models:
+
+- judo-meta-esm
+- judo-meta-pam
+- judo-meta-ui
+
+There are 3 modules in these projects, which are related to the generator commons:
+
+- generator-engine
+- generator-maven-project
+- generator-maven-plugin-test
+
+The `generator-engine` teaches the generator commons how to release the items. This bind the meta model to the generator.
+Models are transformed into template parameters in this project.
+
+
+For example in the judo-meta-esm :
+
+    EsmGenerator.java -> Process of parameterization.
+
+    EsmGeneratorParamete.java -> This contains the model as a parameter, the generatorContext, the DirectoryResolvers etc.
+
+=== Description of a project
+
+The <project type>.yaml file describes what template is used for that and control which parameters are passed and the model can control HOW the templates can be used.
+Each generator-engine project has test directory with test project. This test project provides an example of how a `test-project.yaml` description looks like.
+
+Typically struct is :
+
+[source,yaml]
+----
+templates:
+
+  - name: testOverride
+    pathExpression: "#actorType.name + '/actorToOverride'"
+    templateName: test1/actorToOverride.hbs
+    actorTypeBased: true
+
+  - name: testReplace
+    pathExpression: "#actorType.name + '/actorToReplace'"
+    templateName: test1/actorToReplace.hbs
+    actorTypeBased: true
+
+  - name: testDelete
+    pathExpression: "#actorType.name + '/actorToDelete'"
+    templateName: test1/actorToDelete.hbs
+    actorTypeBased: true
+----
+
+* `templates`: This defines a list of templates.
+
+* `name`: This is like an id for the template. Based on this, we can override the template later.
+
+* `pathExpression`: This specifies where the generated file is located and under what name it will be generated. We use spring expression language (springEL), because handlebars only support string output.
+
+* `templateName`: The location of the handlebars file relative to the root.
+
+* `actorTypeBased`: When actorTypeBased template used, the template called for all actor types and the actorType variable is defined.
+
+[source,java]
+----
+//if the actorTypeBased is true, we need to defined in the EsmGeneratorParameter
+
+    @Builder.Default
+    Predicate<ActorType> actorTypePredicate = a -> true;
+
+    @NonNull
+    Function<ActorType, File> actorTypeTargetDirectoryResolver;
+
+----
+Not in the example :
+
+* `factoryExpression`: Factory expression is used to create files. It returns a list of objects which are used as root context for the given handlebar template.
+
+* `templateContext`: Template context is used to put expression result to template variable. (Not used often because of the helpers)
+
+* `copy`: It can be used to copy a binary file. In this case the template file used as binary, no templating is performed.
+
+* `permission`: It can be used to add execution permission to files. For example bash files.
```

Comment:
Permission format have to be documented. You can check the implemetation. And have to mention that it is based on Posix system, in windows it sets windows's policies because windows does not support that type of permissions

Reply:
[cc88925](https://github.com/BlackBeltTechnology/judo-generator-commons/pull/29/commits/cc88925996d698e19d04ebe7b9327aed6758a460)